core/ppu.hpp

```diff
@@ -22,17 +22,103 @@ class PPU {
   // have to update the buffer when SDL asks for it
   u8* getFrameBuffer();
 
-  // probably a lot of other public interface stuff that you'll discover in implementation
-  // See https://gbdev.io/pandocs/Rendering.html
+  // TODO: decide whether this logic should live in the PPU or if we should have the MMU handle 
+  //       these by allowing it to access the current mode of the PPU
+  // cpuReadVRAM()
+  // cpuWriteVRAM()
+  // cpuReadOAM()
+  // cpuWriteOAM()  
     
 private:
   CPU cpu;
   MMU mmu;
   Mode mode;
 
+  unsigned int cyclesLeft;
+
+  const int LCD_WIDTH = 144;
+  const int LCD_HEIGHT = 160;
+
+  const int OAM_CLOCKS = 80;
+  const int VRAM_CLCOKS = 172;
+  const int HBLANK_CLOCKS = 204;
+  const int VBLANK_CLOCKS = 4560;
+
+  const u16 OAM_TABLE = 0xFE00;
+
+  // LCDC (0xFF40) - LCD Control
+    // Bit 7	LCD and PPU enable	0=Off, 1=On
+    // Bit 6	Window tile map area	0=9800-9BFF, 1=9C00-9FFF
+    // Bit 5	Window enable	0=Off, 1=On
+    // Bit 4	BG and Window tile data area	0=8800-97FF, 1=8000-8FFF
+    // Bit 3	BG tile map area	0=9800-9BFF, 1=9C00-9FFF
+    // Bit 2	OBJ size	0=8x8, 1=8x16
+    // Bit 1	OBJ enable	0=Off, 1=On
+    // Bit 0	BG and Window enable/priority	0=Off, 1=On
+  u16 lcdc = mmu.read(0xFF40);
```

Comment:
At the time of PPU construction, I think this address (and others referenced in this file) will be unitialized (0). Is this something that we have to periodically read?